README.md

```diff
@@ -1,102 +1,189 @@
 
-By default, if none of these parameters are passed, the API is granted - no other rules will be tested, and no permission needs to be checked.
+The configuration files are located in `digital-factory-data/karaf/etc`, with the `org.jahia.modules.api.authorization-*.(yml|cfg)` filename pattern.
+Starting from Jahia 8.1.0.0 you can write configuration in yaml format - for older versions, you must use cfg format. 
 
-Rules can also define matching criterias with the following properties :
+The following snippet :
 
-A rule can specify any number of matching criteria. Each of these criteria can have a single value or a comma separated list of values.
- - `api` : The names of the API, if the rule should only apply to some entry points
- - `pathPattern` : Regular expressions that will be tested on the node path.
- - `workspace` : `live` or `default`, only request on the specified workspace will match.
- - `nodeType` : Only request on nodes of these type will match.
- - `scope` : Only request claiming that scope will match. 
- - `permission` : Only request with users having this permission will match.
- 
-Note the difference between `requiredPermission`/`requiredScope` and `permission`/`scope` - in the "non-required" case, if the user has the permission/scope, his access will be granted (based on `access`, `requiredPermission`, `requiredScope` values), and no other rule will be checked. 
-But it's not required - if he does not have it, the rule won't match, and so subsequent rules will be checked. 
-In the "required" case, if the rule match, the user must have the permission/scope - no other rule will be checked. 
-
-An optional `priority` property can be also specified - the rules with the lowest value will be executed first. Default priority is 0.
-
-```
-permission.checkFirst.priority=-10
+```yaml
+graphql:
+  description: Can access graphql API
+  metadata:
+    visible: true
+  auto_apply:
+    - origin: hosted
+  grants:
+    - api: graphql
+      node: none
+    - api: graphql
+      node:
+        withPermission: api-access
 ```
 
-A rule with no condition will always match (and thus, should have a very high value for priority so that it is checked last):
-```
-permission.global.access=restricted
-permission.global.priority=99999
+will be written this way when using cfg format : 
+```properties
+graphql.description = Can access graphql API
+graphql.metadata.visible = true
+graphql.auto_apply[0].origin = hosted
+graphql.grants[0].api = graphql
+graphql.grants[0].node = none
+graphql.grants[1].api = graphql
+graphql.grants[1].node.withPermission = api-access
 ```
 
-Rules can specify a simple path pattern : 
-```
-permission.users.pathPattern=/sites/[^/]+/home/.*
+Examples below are given in yaml format.
+
+### Scope name, description and metadata
+
+Every scope must have a unique name. The description explains what the scope is granting. 
+Metadata can be freely added and used by UI or other services.
```

Comment:
Can you provide any details on how `metadata` is different from any other random object added by a user? Or do we make is a recommendation/convention, to use `metadata` for any piece of configuration aimed at being interpreted by a service other than security filter ?

In other words, why use:
```
graphql:
  metadata:
    visible: true
```
and not:
```
graphql:
  ui:
    visible: true
```
If at the end security filter do not care about it? Or is there a limitation in accessing anything other than what's under `metadata` ?

Note that I'm not questioning the implementation at all, the objective here is just to see if there are possible improvements/additions to the documentation.

Reply:
metadata is different in the way that it's allowed - graphql/ui/visible is not possible and a config like this will crash. so yes metadata can be used by other services.